Albums/Albums/ViewControllers/AlbumDetailTableViewController.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  AlbumDetailTableViewController.swift
+//  Albums
+//
+//  Created by Cora Jacobson on 7/16/20.
+//  Copyright © 2020 Cora Jacobson. All rights reserved.
+//
+
+import UIKit
+
+class AlbumDetailTableViewController: UITableViewController {
+    
+    var albumController: AlbumController?
+    
+    var album: Album? {
+        didSet {
+            updateViews()
+        }
+    }
+    
+    var tempSongs: [Song] = []
+    
+    @IBOutlet weak var albumNameTextField: UITextField!
+    @IBOutlet weak var artistTextField: UITextField!
+    @IBOutlet weak var genresTextField: UITextField!
+    @IBOutlet weak var coverURLsTextField: UITextField!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        guard isViewLoaded else { return }
+        updateViews()
+    }
+
+    // MARK: - Table view data source
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return tempSongs.count + 1
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "songCell", for: indexPath) as? SongTableViewCell else { return UITableViewCell() }
+        cell.delegate = self
+        switch indexPath.row {
+        case tempSongs.count:
+            return cell
+        default:
+            cell.song = tempSongs[indexPath.row]
+            return cell
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        if indexPath.row < tempSongs.count {
+            return 100
+        } else {
+            return 130
+        }
+    }
+    
+    @IBAction func saveButton(_ sender: UIBarButtonItem) {
+        if let name = albumNameTextField.text,
+            !name.isEmpty,
+            let artist = artistTextField.text,
+            !artist.isEmpty,
+            let genres = genresTextField.text,
+            !genres.isEmpty,
+            let coverArt = coverURLsTextField.text,
+            !coverArt.isEmpty,
+            let albumController = albumController {
+            if let album = album {
+                albumController.update(album: album, name: name, artist: artist, genres: genres.components(separatedBy: ", "), coverArt: coverArt.components(separatedBy: ", "), songs: tempSongs)
+            } else {
+                albumController.createAlbum(name: name, artist: artist, id: UUID().uuidString, genres: genres.components(separatedBy: ", "), coverArt: coverArt.components(separatedBy: ", "), songs: tempSongs)
+            }
+            DispatchQueue.main.async {
+                albumController.albums = []
+                albumController.getAlbums { (result) in
+                    DispatchQueue.main.async {
+                        print("Refreshed data")
+                    }
+                }
```

Comment:
These lines are not needed. When you pop to the last view controller, the data should be reloaded in your viewWillAppear. As it is, adding a song pops back to the tableView to show no albums.